stats/stats.go

```diff
@@ -78,6 +78,13 @@ func StartStatsServerWithPort(startupOpts StartupOptions, addhandlers ...func(mu
 			setLevel = zapcore.InfoLevel
 		}
 		logLevel = zap.NewAtomicLevelAt(setLevel)
+
+		logConfig := zap.NewProductionConfig()
+		logConfig.Level = logLevel
+		logger, logErr := logConfig.Build()
+		if logErr == nil {
+			contextutils.SetFallbackLogger(logger.Sugar())
+		}
 	} else if startupOpts.LogLevel != nil {
 		logLevel = *startupOpts.LogLevel
```

Comment:
Is there a reason this case doesn't need the same `zap.NewProductionConfig` lines that were added?

Reply:
I checked Gloo for this as well. Looks like the accesslogger component is the only thing that is setting a loglevel through the startup options. In that case, it is also creating it's own logger in `contextutils` so I'm guessing if we define it this way then it is expected that the logger is set outside of this scope.